deep_qa/common/params.py

```diff
@@ -83,42 +81,41 @@ def pop_choice(self, key: str, choices: List[Any], name: str=None) -> Union["Par
 
         value = self.pop(key)
         if value not in choices:
-            raise ConfigurationError(self._get_choice_error_message(value, choices, name))
-        return self.__check_is_dict(key, value)
+            raise ConfigurationError(_get_choice_error_message(value, choices, self.history))
+        return value
 
     def pop_choice_with_default(self,
                                 key: str,
                                 choices: List[Any],
-                                default: Any=None,
-                                name: str=None) -> Union["Params", Any]:
+                                default: Any=None):
         """
         Like get_choice, but with a default value.  If `default` is None, we use the first item in
         `choices` as the default.
         """
         try:
-            return self.pop_choice(key, choices, name)
+            return self.pop_choice(key, choices)
         except KeyError:
             if default is None:
                 default = choices[0]
             if default not in choices:
-                raise ConfigurationError(self._get_choice_error_message(default, choices, name))
+                raise ConfigurationError(_get_choice_error_message(default, choices, self.history))
 
             return self.__check_is_dict(key, default)
 
+    def as_dict(self):
+        """
+        Sometimes we need to just represent the parameters as a dict, for instance when we pass
+        them to a Keras layer(so that they can be serialised).
+        """
+        return self.params
```

Comment:
Can you do a recursive `logger.param()` on everything in here, before returning it, as well as an initial statement like `logger.param("Converting param object to dictionary; logging of default values won't work with the result")`?